runtime/databricks/automl_runtime/forecast/pmdarima/training.py

```diff
@@ -65,25 +66,51 @@ def fit(self, df: pd.DataFrame) -> pd.DataFrame:
         # Impute missing time steps
         history_pd = self._fill_missing_time_steps(history_pd, self._frequency_unit)
 
+        history_timedelta = history_pd['ds'].max() - history_pd['ds'].min()
+        timedeltas = history_pd['ds'].diff()
+        min_timedelta = timedeltas.iloc[timedeltas.values.nonzero()[0]].min()
+
         # Tune seasonal periods
         best_result = None
         best_metric = float("inf")
+        all_validation_horizons = dict()
+        all_cutoffs = dict()
         for m in self._seasonal_periods:
             try:
-                cutoffs = generate_cutoffs(history_pd, horizon=self._horizon, unit=self._frequency_unit,
-                                           num_folds=self._num_folds, seasonal_period=m)
+                # this check mirrors the the default behavior by prophet
+                seasonality_timedelta = pd.to_timedelta(m, unit=self._frequency_unit)
+                if history_timedelta < 2 * seasonality_timedelta:
+                    _logger.warning(f"Skipping seasonal_period={m} ({seasonality_timedelta}). Dataframe timestamps must span at least two seasonality periods, but only spans {history_timedelta}")
+                    continue
+                # Prophet also rejects the seasonality periods if the seasonality period timedelta is less than the shortest timedelta in the dataframe.
+                # However, this cannot happen in ARIMA because _fill_missing_time_steps imputes values for each _frequency_unit,
+                # so the minimum valid seasonality period is always 1
+
+                validation_horizon = utils.get_validation_horizon(history_pd, self._horizon, self._frequency_unit)
```

Comment:
Yeah I think that might be helpful. `ArimaEstimator` and `ProphetHyperoptEstimator` are pretty different since Arima loops through seasonalities, while Prophet uses hyperopt, but it would still be cleaner to put the common code in a base class. Filed https://databricks.atlassian.net/browse/ML-22554